src/wasm/wasm-validator.cpp

```diff
@@ -796,6 +807,37 @@ void FunctionValidator::visitCall(Call* curr) {
     return;
   }
   validateCallParamsAndResult(curr, target->type);
+
+  if (Intrinsics(*getModule()).isCallWithoutEffects(curr)) {
+    // call.without.effects has the specific form of the last argument being a
+    // function reference, which will be called with all the previous arguments.
+    // The type must be consistent with that. This, for example, is not:
+    //
+    //  (call $call.without.effects
+    //    (i32.const 1)
+    //    (.. some function reference that expects an f64 param and not i32 ..)
+    //  )
+    if (!curr->operands.empty()) {
+      auto* target = curr->operands.back();
+      // Ignore unreachable here, and leave other validation issues here for the
+      // global scope which already does this.
```

Comment:
Not sure what this means. What are the other validation issues? And what do we need to handle if this is unreachable?

Reply:
Yes, this wasn't very clear... I rewrote it not.

If it's unreachable we can ignore it. If it's not a function, then that's a problem with the signature of the import which has the wrong form, and we have validation for that already.